notebooks/6_Putting_it_All_Together.ipynb

```diff
@@ -0,0 +1,1002 @@
+      },
+      "text/plain": [
+       "HBox(children=(HTML(value=''), FloatProgress(value=0.0, max=177.0), HTML(value='')))"
+      ]
+     },
+     "metadata": {},
+     "output_type": "display_data"
+    },
+    {
+     "name": "stdout",
+     "output_type": "stream",
+     "text": [
+      "\n"
+     ]
+    }
+   ],
+   "source": [
+    "from skcosmo.selection import SampleCUR\n",
+    "\n",
+    "num_active = int(len(i_train) / 10)\n",
+    "sample_mixing = 0.5\n",
+    "\n",
+    "print(f\"Selecting {num_active} samples\")\n",
+    "scur = SampleCUR(X=X_train, Y=Yp_train, mixing=sample_mixing, progress_bar=True)\n",
+    "sample_idx = scur.select(num_active)\n",
+    "\n",
+    "X_active = X_train[sample_idx]"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 21,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "# np.savez('active_set_file.npz', idx=sample_idx)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "### Skip Choosing Active Set"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 22,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "# # Import active set indices from numpy file\n",
+    "\n",
+    "# index_file = 'active_set_file.npz'\n",
+    "# data = np.load(index_file)\n",
+    "# sample_idx = data['idx']\n",
+    "\n",
+    "# X_active = X_train[sample_idx]"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "# Constructing a Sparse KPCovR"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 23,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "from sklearn.metrics.pairwise import pairwise_kernels\n",
+    "from scipy.linalg import svd\n",
+    "\n",
+    "from skcosmo.pcovr import KPCovR\n",
+    "from matplotlib import pyplot as plt\n",
+    "\n",
+    "kernel = \"rbf\"\n",
+    "kernel_params = {\"gamma\": 0.2}"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## Determine the active kernel\n",
+    "**Keep in mind that this example is originally written with an environment-centred property, and therefore we can use an environment kernel.**\n",
+    "\n",
+    "Until you can specify an active set with sklearn, we'll construct a sparse nystroem approximation in a similar manner, and replace this with the sklearn functionality when it becomes available."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 24,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "active_kernel = pairwise_kernels(\n",
+    "    X_active, metric=kernel, filter_params=True, **kernel_params\n",
+    ")\n",
+    "\n",
+    "# By including the right singular vectors into the RKHS features, we normalize and center the RKHS.\n",
+    "U, S, V = svd(active_kernel)\n",
+    "S = np.maximum(S, 1e-12)\n",
+    "normalization_ = np.dot(U / np.sqrt(S), V)\n",
+    "components_ = X_active.copy()\n",
+    "\n",
+    "Phi_train = np.dot(\n",
```

Comment:
This part of the code is not very clear. Amongst other things, Phi is not defined, the `1e-12` in `S = np.maximum(S, 1e-12)`  is a bit arbitrary, and the use of SVD to build a normalization matrix is not explained. Is this talked about in the previous notebooks?